### PR TITLE
Enable OV clang format command for nGraph

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Reuse OV cmake scripts to enable clang-format code style check
     - `style-check` and `style-apply` commands were removed
     - Added custom targets to check code style.
     - Main targets for clang-format are `clang_format_check_all` and `clang_format_fix_all`

### CODE REVIEW

File: .github/workflows/code_style.yml [hunk 1]
```diff
@@ -24,20 +24,19 @@ jobs:
...
-        working-directory: build
+        run: cmake --build build --target clang_format_check_all
 
       - name: Create code style diff
         if: failure()
         run: |
-          ngraph/maint/apply-code-format.sh
-          git diff >ngraph_code_style_diff.patch
+          cmake --build build --target clang_format_fix_all
+          git diff > code_style_diff.patch
```
Comment (t-jankowski) [inline .github/workflows/code_style.yml:33]: I'd suggest to use `.diff` file extention. A patch is an output of `git format-patch` command and adds additional meta-data to _regular_ diff output.

File: .github/workflows/code_style.yml [hunk 2]
```diff
@@ -24,20 +24,19 @@ jobs:
...
 
       - uses: actions/upload-artifact@v2
         if: failure()
         with:
-          name: ngraph_code_style_diff
-          path: ngraph_code_style_diff.patch
+          name: code_style_diff
+          path: code_style_diff.patch
```
Comment (t-jankowski) [inline .github/workflows/code_style.yml:39]: As above - `.diff` vs `.patch`

**Review by t-jankowski — APPROVED** [review]

File: .github/workflows/code_style.yml (hunk 2 above)
Comment (ilyachur) [reply .github/workflows/code_style.yml:39]: It is a legacy extension 😄 But yes, I agree that we can rename it.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by postrational** [comment]
It seems that this approach makes it impossible to build nGraph if there is any error in whitespace. This will make development tedious, as you will have to make sure your formatting is correct at every step, not only in your final PR.

**Comment by ilyachur** [comment]
> It seems that this approach makes it impossible to build nGraph if there is any error in whitespace. This will make development tedious, as you will have to make sure your formatting is correct at every step, not only in your final PR.

Yes I integrated the code-style check to the build process as it is already working for Inference Engine and cpplint.
Let's discuss this topic, it is really easy to disable code style check for the build process. 

Honestly we didn't get a negative feedback for IE (but for IE the code style check also is integrated to the build) and as I know the clang-format is supported by majority code editors it means that editors can fix code style automatically before the file saving.

Let's discuss it. I am opened for any feedback and I can disable this check if it is uncomfortable.

File: .github/workflows/code_style.yml (hunk 2 above)
Comment (ilyachur) [reply .github/workflows/code_style.yml]: Fixed

**Comment by ilyachur** [comment]
@postrational I disabled code-style check for compilation.

**Review by jdanieck — APPROVED** [review]

**Review by tsocha — APPROVED** [review]

**Review by pelszkow — APPROVED** [review]

**Review by tomdol — APPROVED** [review]

**Review by mitruska — APPROVED** [review]

**Review by ababushk — APPROVED** [review]

**Pull request merged**